src/constants/environment.ts

```diff
@@ -63,6 +63,8 @@ export const IS_CHROME_BASED = userAgentLowerCase.includes('chrome');
 
 export const IS_FIREFOX = userAgentLowerCase.includes('firefox');
 
+export const IS_SAFARI = /Mozilla\/5.0 \((Macintosh|iPhone); [\s\S]+?\) AppleWebKit\/\S+ \(KHTML, like Gecko\)( (Version|Safari|Mobile)\/\S+)+/.test(navigator.userAgent);
```

Comment:
I'm not sure how `load` event behaves in other environments, I've just took `navigator.userAgent` of that I have
- "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/17.4.1 Safari/605.1.15"
- "Mozilla/5.0 (iPhone; CPU iPhone OS 17_2 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Mobile/15E148" (Simulator)
- "Mozilla/5.0 (iPhone; CPU iPhone OS 17_4_1 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Mobile/15E148" (Real device)

also, this query shouldn't overlap with other browsers because it actually disables functionality.